src/asm/section.c

```diff
@@ -416,6 +428,35 @@ void out_EndLoadSection(void)
 	loadOffset = 0;
 }
 
+void out_PushInlineFragmentSection(void)
+{
+	checkcodesection();
+
+	if (currentLoadSection)
+		fatalerror("`LOAD` blocks cannot contain inline fragments\n");
+
+	struct Section *sect = pCurrentSection;
+
+	// A section containing an inline fragment has to become a fragment too
+	sect->modifier = SECTION_FRAGMENT;
```

Comment:
What if it's already a `SECTION UNION`? I guess it shouldn't happen, given that the only two contexts they're currently allowed to appear in are only valid for data sections, which cannot be `UNION`s. But both might possibly change in the future, so I'd expect at least a test case.

Reply:
I wasn't expecting `SECTION UNION` to ever be valid for ROM sections, but sure.

Reply:
My point is that it *currently* isn't, but that might be subject to change in the future (especially if we start supporting more exotic cartridge configurations).

Reply:
Okay. For now I'll disallow inline fragments in `SECTION UNION`, to revisit if that does change.